molgenis-data-import/src/main/java/org/molgenis/data/importer/IntermediateParseResults.java

```diff
@@ -107,8 +108,7 @@ public EntityMetaData addEntityMetaData(String name)
 			}
 		}
 
-		EntityMetaData emd = new EntityMetaData(simpleName,
-				getApplicationContext().getBean(EntityMetaDataMetaData.class));
+		EntityMetaData emd = new EntityMetaData(simpleName, applicationContext.getBean(EntityMetaDataMetaData.class));
```

Comment:
Wouldn't it be better to use entityMetaDataFactory.create() instead and prevent injection of ApplicationContext?